src/snowflake/snowpark/modin/pandas/general.py

```diff
@@ -1786,20 +1785,134 @@ def crosstab(
     """
     Compute a simple cross tabulation of two (or more) factors.
     """
-    # TODO: SNOW-1063345: Modin upgrade - modin.pandas functions in general.py
-    pandas_crosstab = pandas.crosstab(
-        index,
-        columns,
-        values,
-        rownames,
-        colnames,
-        aggfunc,
-        margins,
-        margins_name,
-        dropna,
-        normalize,
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
+
+    if not is_nested_list_like(index):
+        index = [index]
+    if not is_nested_list_like(columns):
+        columns = [columns]
+
+    from pandas.core.reshape.pivot import _build_names_mapper, _get_names
+
+    rownames = _get_names(index, rownames, prefix="row")
+    colnames = _get_names(columns, colnames, prefix="col")
+
+    (
+        rownames_mapper,
+        unique_rownames,
+        colnames_mapper,
+        unique_colnames,
+    ) = _build_names_mapper(rownames, colnames)
+
+    common_idx = None
+    pass_objs = [x for x in index + columns if isinstance(x, (Series, DataFrame))]
+    if pass_objs:
+        if len(pass_objs) == 1:
+            common_idx = pass_objs[0]
+        else:
+            common_idx = pass_objs[0].intersection(pass_objs[1:])
+
+    data = {
+        **dict(zip(unique_rownames, index)),
+        **dict(zip(unique_colnames, columns)),
+    }
+    df = DataFrame(data, index=common_idx)
+
+    if values is None:
+        df["__dummy__"] = 0
+        kwargs = {"aggfunc": "count"}
+    else:
+        df["__dummy__"] = values
+        kwargs = {"aggfunc": aggfunc}
+
+    table = df.pivot_table(
```

Comment:
Can we also pass the calling method name (`crosstab` in this case) so that the NotImplementedError messages that would be raised inside `pivot_table` show the correct method name that was originally called by the user?

Reply:
(For less confusion and easier debuggability)

Reply:
I"m not certain that that's possible? Unless we catch the not implemented error and edit it? There really shouldn't be any NotImplementedError though, since all of the arguments of crosstab are supported.

Reply:
There can be unsupported aggregation functions, like `median` (we should test that, by the way). I think that it's ok if the error message mentions `pivot_table`. I think the user will be able to understand that error message, and our telemetry will tell us that the original function was `crosstab`.

Reply:
I don't think it will be clear to the user, especially if it's part of a chain of operations. The chain may even include an actual `pivot_table` operation that is well supported. The error message in this case will be very confusing.

Reply:
This is a potential problem with many cases where Snowpark pandas methods call other Snowpark pandas methods. In my opinion it's not worth fixing.

Reply:
> There can be unsupported aggregation functions, like median (we should test that, by the way). I think that it's ok if the error message mentions pivot_table. I think the user will be able to understand that error message, and our telemetry will tell us that the original function was crosstab.

Good catch - completely slipped my mind! Will add a test for unsupported aggs!

Reply:
You're right. We indeed frequently have Snowpark pandas methods call other Snowpark pandas methods. We did take care of passing the calling method name as an argument in some cases to customize the error messages accordingly. However, I'm fine with not worrying about this for the purpose of this PR, and possibly handling all similar cases in the code base in one pass at some point.